### DISCUSSION TITLE
Question: how much storage space does a null value occupy?

### DISCUSSION THREAD

**Discussion by david542542** [duckdb/duckdb#7248] [Q&A]
This is a bit different between different RDBMs as to how much space a `null` value occupies. For example, some use a bitmap for storing nulls, other use the column width if a fixed size, and others somewhere in between.

In DuckDB, how much storage does a `null` value occupy?

**Comment by Mytherin** [comment]
DuckDB stores nulls using a bitmask - but also keeps gaps around in the original columns for `NULL` values. In the uncompressed case this is therefore the column width. However, because of compression, the `NULL` values take up almost no space if there are many clustered `NULL` values (or few/none).
